src/program.rs

```diff
@@ -304,10 +304,20 @@ impl<EC: EvalCache> Program<EC> {
         String::from_utf8(buffer.into_inner().into_inner()).unwrap()
     }
 
-    /// Create a markdown file with documentation for the specified program in `.nickel/doc/program_main_file_name.md`
+    /// Extract documentation from the program
     #[cfg(feature = "doc")]
-    pub fn output_doc(&mut self, out: &mut dyn std::io::Write) -> Result<(), Error> {
-        doc::output_doc(self.vm.import_resolver_mut(), self.main_id, out)
+    pub fn extract_docs(&mut self) -> Result<doc::ExtractedDocumentation, Error> {
```

Comment:
```suggestion
    pub fn extract_doc(&mut self) -> Result<doc::ExtractedDocumentation, Error> {
```

Nitpick, but I think we use the singular form otherwise as in `ExtractedDocumentation`.